### PR TITLE
feat(`forge selectors`): add `find` command

### LINKED ISSUE #8726: feat(`forge selectors`): add option to `find` a selector in the given codebase
### Component

Forge

### Describe the feature you would like

When building on top of other protocols, devs sometimes encounter reverts due to unknown selectors. If a selector isn’t listed in a registry like [4bytes.directory](4bytes.directory), we need to manually search for it by cloning the project, running `forge selectors list`, and finding the error. For large projects, `forge selectors list` output can be too long and get cut off in the terminal. 

To make this easier, I suggest adding a `find` option to the `forge selectors` command. 

**Example:**

```
forge selectors find 0x154c51b8
```

`0x154c51b8` is a selector not present in the registry. The `find` option would help us quickly locate the function/error.

I'm happy to implement this! lmk wdyt.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PraneshASP**
## Description:
- Closes #8726 

## Usage:
```sh
forge selectors find 0xdd62ed3e 
```

#### Output:
Screenshot 2024-08-27 at 7 48 49 PM

Supports partial inputs as well:
```sh
forge selectors find dd  # lists all functions/events/errors with selector starting with `dd`
```

#### Output:
Screenshot 2024-08-27 at 7 53 34 PM

### CODE REVIEW

**Review by zerosnacks — APPROVED** [review]
Thanks @PraneshASP this is great

File: crates/forge/bin/cmd/selectors.rs [hunk 1]
```diff
@@ -260,6 +271,81 @@ impl SelectorsSubcommands {
...
+                            ]);
+                        }
+                    }
+                }
+
+                if table.row_count() > 0 {
+                    println!("Found {} instance(s)...", table.row_count());
+                    println!("{table}");
+                } else {
+                    println!("Selector not found in the project.");
```
Comment (DaniPopes) [inline crates/forge/bin/cmd/selectors.rs:346]: maybe error here?

Comment (PraneshASP) [reply crates/forge/bin/cmd/selectors.rs]: Updated: https://github.com/foundry-rs/foundry/pull/8754/commits/96fee621d3fe22419c4b90b5dc06b5624cb0d105

**Review by DaniPopes — APPROVED** [review]
lgtm

**Pull request merged**